MANUAL.md

```diff
@@ -0,0 +1,211 @@
+
+```rust
+let mut testkit = TestKitBuilder::validator()
+    .with_service(MyService::new())
+    .with_service(OtherService::new())
+    .create();
+// Create transactions.
+let tx1 = MyTransaction::new(...);
+let tx2 = OtherTransaction::new(...);
+// Commit its into blockchain.
+testkit.create_block_with_transactions(txvec![tx1, tx2]);
+// Check the expected result.
+```
+
+### API testing
+
+The following steps may help you.
+
+* Define the `MyServiceApi` trait for the `TestKitApi` structure that covers the whole API of your service.
+* Implement functions that use transactions to fill your storage with the test data.
+* Create the tests that check all of your endpoints.
+
+```rust
+
+// API trait definition.
+
+trait MyServiceApi {
+    fn get_public_data(&self) -> ApiResponsePublicData;
+    fn get_private_data(&self) -> ApiResponsePrivateData;
+    fn post_private_data(&self, data: MyPrivateData) -> ApiResponsePostPrivateData;
+}
+
+impl MyServiceApi for TestKitApi {
+    fn get_public_data(&self) -> ApiResponsePublicData {
+        self.get(ApiKind::Service("my_service"), "/v1/first_endpoint")
+    }
+
+    fn get_private_data(&self) -> ApiResponsePublicData {
+        self.get_private(ApiKind::Service("my_service"), "/v1/second_endpoint")
+    }
+
+    fn post_private_data(&self, data: &MyPrivateData) -> ApiResponsePostPrivateData {
+        self.post(
+            ApiKind::Service("my_service"),
+            "v1/third_endpoint",
+            &data,
+        )
+    }
+}
+
+// API test skeleton
+
+#[test]
+fn my_api_test() {
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    fill_storage_with_data(&mut testkit);
+    // Check API responses
+    let api = testkit.api();
+    assert_eq!(api.get_public_data(), ApiResponsePublicData::new(...));
+    ...
+}
+```
+
+## Advanced usage
+
+Here are examples of more complex and less common cases.
+
+### Oracles testing
+
+The oracle in this case is a service which can produce transactions with external data after the commit of the block,
+[`exonum-time`][exonum-time] and [`exonum-btc-anchoring`][exonum-btc-anchoring] are examples of these kind of oracles.
+In this way, transactions created during the `handle_commit` execution will be stored in `TestKit` memory pool.
+
+```rust
+// Create testkit with the service which creates transaction with the height
+// of latest committed block after commit.
+let mut testkit = TestKitBuilder::validator()
+    .with_service(HandleCommitService)
+    .create();
+// Call the `handle_commit` event.
+testkit.create_block();
+// Check that `handle_commit` has been invoked at the correct height.
+let tx = TxAfterCommit::new_with_signature(Height(1), &Signature::zero());
+assert!(testkit.mempool().contains_key(&tx.hash()));
+```
+
+In order to invoke a `handle_commit` event, you must create a block.
+If the oracle has to fetch any data from external world, you must create a mock object.
```

Comment:
"you must create a mock object." I would clarify a bit more like "you must create a mock object that would generate said external data to accomplish testing"